corehq/apps/consumer_user/views.py

```diff
@@ -0,0 +1,195 @@
+        return {}
+
+    def get_success_url(self):
+        url = self.get_redirect_url()
+        if url:
+            return url
+        url = reverse('consumer_user:consumer_user_homepage')
+        return url
+
+    def get_context_data(self, **kwargs):
+        context = super(ConsumerUserLoginView, self).get_context_data(**kwargs)
+        if self.hashed_invitation:
+            extra_context = {}
+            this_is_not_me = reverse('consumer_user:consumer_user_register',
+                                     kwargs={
+                                         'invitation': self.hashed_invitation
+                                     })
+            extra_context['this_is_not_me'] = '%s%s' % (this_is_not_me,
+                                                        '?create_user=1')
+            context.update(extra_context)
+        context.update({'hide_menu': True})
+        return context
+
+
+def get_invitation_object_or_error(invitation):
+    try:
+        decoded_invitation = urlsafe_base64_decode(TimestampSigner().unsign(invitation,
+                                                                            max_age=timedelta(days=30)))
+        invitation_obj = ConsumerUserInvitation.objects.get(id=decoded_invitation)
+        if invitation_obj.accepted:
+            url = reverse('consumer_user:consumer_user_login')
+            return HttpResponseRedirect(url)
+        elif not invitation_obj.active:
+            return HttpResponse(_("Invitation is inactive"), status=400)
+        else:
+            return invitation_obj
+    except (BadSignature, ConsumerUserInvitation.DoesNotExist):
+        return HttpResponse(_("Invalid invitation"), status=400)
+    except SignatureExpired:
+        return HttpResponse(_("Invitation is expired"), status=400)
+
+
+@two_factor_exempt
+def register_view(request, invitation):
+    invitation_obj = get_invitation_object_or_error(invitation)
+    if isinstance(invitation_obj, HttpResponse):
+        return invitation_obj
+    email = invitation_obj.email
+    create_user = request.GET.get('create_user', False)
+    if create_user != '1' and User.objects.filter(username=email).exists():
+        url = reverse('consumer_user:consumer_user_login_with_invitation',
+                      kwargs={
+                          'invitation': invitation
+                      })
+        return HttpResponseRedirect(url)
+    if request.method == "POST":
+        form = ConsumerUserSignUpForm(request.POST, invitation=invitation_obj)
+        if form.is_valid():
+            form.save()
+            url = reverse('consumer_user:consumer_user_login')
+            return HttpResponseRedirect(url)
+    else:
+        form = ConsumerUserSignUpForm()
+    return render(request, 'signup.html', {'form': form,
+                                           'hide_menu': True})
+
+
+@two_factor_exempt
+@sensitive_post_parameters('auth-password')
+def login_view(request, invitation=None):
```

Comment:
Why use this functional view AND a class-based view?  Couldn't you do all this in the `dispatch` method of `ConsumerUserLoginView`?